src/types/router.rs

```diff
@@ -164,6 +165,34 @@ pub mod engine {
             }
         }
 
+        /// Compute the total Haversine distance of a path.
+        ///
+        /// # Arguments
+        /// * `path` - The path to compute the distance of. The path is
+        ///   given as a vector of [`NodeIndex`] structs.
+        ///
+        /// # Returns
+        /// The total distance of the path.
+        ///
+        /// If the path is empty, 0.0 is returned.
+        ///
+        /// If the path is invalid, -1.0 is returned.
+        pub fn get_total_distance(&self, path: &Vec<NodeIndex>) -> f32 {
```

Comment:
Does it make more sense to return a Result<f32, Error> instead of returning -1 for invalid paths, then the caller can discard error types? A bit semantic but I get nervous seeing floating arbitrary values/magic numbers

Reply:
[7ee52f0](https://github.com/Arrow-air/lib-router/pull/11/commits/7ee52f056943f44a59510c46820108ae8217fa8c)